lib/rules/selector-disallowed-list/index.cjs

```diff
@@ -75,38 +75,38 @@ const rule = (primary, secondaryOptions) => {
 				}
 			}
 
-			if (splitList) {
-				parseSelector(getRuleSelector(ruleNode), result, ruleNode)?.each((selector) => {
-					const { selector: selectorStr, index, endIndex } = getStrippedSelectorSource(selector);
-
-					if (matchesStringOrRegExp(selectorStr, primary)) {
-						report({
-							result,
-							ruleName,
-							message: messages.rejected,
-							messageArgs: [selectorStr],
-							node: ruleNode,
-							index,
-							endIndex,
-						});
-					}
-				});
-			} else {
-				const { selector, raws } = ruleNode;
-
-				if (matchesStringOrRegExp(selector, primary)) {
-					const word = (raws.selector && raws.selector.raw) || selector;
-
+			/**
+			 * @param {string} selectorStr
+			 * @param {number} index
+			 * @param {number} endIndex
+			 */
+			function check(selectorStr, index, endIndex) {
+				if (matchesStringOrRegExp(selectorStr, primary)) {
 					report({
 						result,
 						ruleName,
 						message: messages.rejected,
-						messageArgs: [selector],
+						messageArgs: [selectorStr],
 						node: ruleNode,
-						word,
+						index,
+						endIndex,
 					});
 				}
 			}
+
+			const selectors = getRuleSelector(ruleNode);
```

Comment:
nit

We have 3 cases of singular and one plural.
i.e. maybe rename `selector`